src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,191 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private void validIndexCheck(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Invalid index: " + index);
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(null, value, null);
```

Comment:
When creating a new `Node<T>`, pass all the fields to the `Node` constructor. It looks like you have created a new node but then you are creating another new node at line 36 which is redundant.